wagtailcache/models.py

```diff
@@ -0,0 +1,92 @@
+from typing import List
+
+from django.core.cache import caches
+from django.db import models
+from django.db.models import Q
+from django.utils.timezone import now
+
+from wagtailcache.settings import wagtailcache_settings
+from wagtailcache.utils import batched
+
+
+class KeyringItemManager(models.Manager):
+    def __init__(self):
+        super().__init__()
+        self._wagcache = caches[wagtailcache_settings.WAGTAIL_CACHE_BACKEND]
+
+    def set(self, url, key, expiry) -> "KeyringItem":
+        """
+        Create or update a keyring item, clearing expired items too.
+        """
+        item, _ = self.update_or_create(
+            defaults={"expiry": expiry},
+            url=url,
+            key=key,
+        )
+        self.clear_expired()
+        return item
+
+    def bulk_delete_cache_keys(self, keys: List[str]) -> None:
```

Comment:
Would prefer if keys is a QuerySet. Because during the delete, we are normally querying keys to delete, flattening them as values list, then passing it in here and re-querying. Double hits to the database. Would prefer to only make one database hit.